src/arrayMethodSort.js

```diff
@@ -4,9 +4,39 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function(compareFunction = compareAsStrings) {
+    let count;
+
+    do {
+      count = 0;
+
+      for (let i = 1; i < this.length; i++) {
+        const previousValue = this[i - 1];
+        const currentValue = this[i];
+
+        if (compareFunction(previousValue, currentValue) > 0) {
+          this[i - 1] = currentValue;
+          this[i] = previousValue;
+          count++;
+        }
+      }
+    } while (count > 0);
+
+    return this;
   };
 }
 
+function compareAsStrings(a, b) {
+  const stringA = String(a);
+  const stringB = String(b);
+
+  if (stringA > stringB) {
+    return 1;
+  } else if (stringA < stringB) {
+    return -1;
+  } else {
+    return 0;
+  }
```

Comment:
```suggestion
  if (stringA > stringB) {
    return 1;
  }
  
  if (stringA < stringB) {
    return -1;
  }
  
  return 0;
```

Reply:
If an `if` block always executes a `return` statement, the subsequent `else` block is unnecessary. A `return` in an `else if` block following an `if` block that contains a `return` can be separated into multiple `if` blocks

More details here
https://mate-academy.github.io/style-guides/javascript.html#blocks--no-else-return